packages/react-data-grid/src/Canvas.tsx

```diff
@@ -0,0 +1,398 @@
+  rowHeight: number;
+  height: number;
+  width?: number;
+  totalWidth?: number | string;
+  style?: string;
+  className?: string;
+  rowOverscanStartIdx: number;
+  rowOverscanEndIdx: number;
+  rowVisibleStartIdx: number;
+  rowVisibleEndIdx: number;
+  colVisibleStartIdx: number;
+  colVisibleEndIdx: number;
+  colOverscanStartIdx: number;
+  colOverscanEndIdx: number;
+  rowsCount: number;
+  rowGetter: RowGetter;
+  columns: ColumnList;
+  cellMetaData: CellMetaData;
+  selectedRows?: RowData[];
+  rowKey: string;
+  scrollToRowIndex?: number;
+  contextMenu?: React.ReactElement;
+  rowSelection?: { indexes?: number[] } | { isSelectedKey?: string } | { keys?: { values: unknown[]; rowKey: string } };
+  rowGroupRenderer?: React.ComponentType;
+  isScrolling: boolean;
+  length?: number;
+  enableCellSelect: boolean;
+  enableCellAutoFocus: boolean;
+  cellNavigationMode: CellNavigationMode;
+  eventBus: EventBus;
+  editorPortalTarget: Element;
+  RowsContainer: React.ComponentType<RowsContainerProps>;
+  lastFrozenColumnIndex?: number;
+  totalColumnWidth: number;
+  interactionMasksMetaData: InteractionMasksMetaData;
+  onScroll(position: ScrollPosition): void;
+  getSubRowDetails?(rowIdx: RowData): SubRowDetails;
+}
+
+type RendererProps = Pick<Props, 'rowVisibleStartIdx' | 'rowVisibleEndIdx' | 'columns' | 'cellMetaData' | 'colVisibleStartIdx' | 'colVisibleEndIdx' | 'colOverscanEndIdx' | 'colOverscanStartIdx' | 'lastFrozenColumnIndex' | 'isScrolling'> & {
+  ref(row: (RowRenderer & React.Component<RowRendererProps>) | null): void;
+  key: number;
+  idx: number;
+  row: RowData;
+  subRowDetails?: SubRowDetails;
+  height: number;
+  isSelected: boolean;
+  scrollLeft: number;
+};
+
+export default class Canvas extends React.PureComponent<Props> {
+  static displayName = 'Canvas';
+
+  static defaultProps = {
+    RowsContainer
+  };
+
+  private readonly canvas = React.createRef<HTMLDivElement>();
+  private readonly interactionMasks = React.createRef<InteractionMasks>();
+  private readonly rows = new Map<number, RowRenderer & React.Component<RowRendererProps>>();
+  unsubscribeScrollToColumn?(): void;
+  _scroll = { scrollTop: 0, scrollLeft: 0 };
+
+  componentDidMount() {
+    this.unsubscribeScrollToColumn = this.props.eventBus.subscribe(EventTypes.SCROLL_TO_COLUMN, this.scrollToColumn);
+  }
+
+  componentWillUnmount() {
+    this._scroll = { scrollTop: 0, scrollLeft: 0 };
+    this.rows.clear();
+    this.unsubscribeScrollToColumn!();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const { scrollToRowIndex } = this.props;
+    if (scrollToRowIndex && prevProps.scrollToRowIndex !== scrollToRowIndex) {
+      this.scrollToRow(scrollToRowIndex);
+    }
+  }
+
+  handleScroll = (e: React.UIEvent<HTMLDivElement>) => {
+    if (this.canvas.current !== e.target) {
+      return;
+    }
+    const { scrollLeft, scrollTop } = e.target as HTMLElement;
+    const scroll = { scrollTop, scrollLeft };
+    this._scroll = scroll;
```

Comment:
I tnohink we can probably remove `this._scroll` and just use props as we anyways call `this.props.onScroll`. I will clean it up in separate PR